src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventServiceImpl.java

```diff
@@ -49,32 +60,78 @@ public Event createEvent(EventDTO eventDto) throws CTPException {
           CTPException.Fault.RESOURCE_NOT_FOUND,
           String.format(
               "Collection exercise %s does not exist", eventDto.getCollectionExerciseId()));
-    } else {
-      Event existing =
-          this.eventRepository.findOneByCollectionExerciseAndTag(collex, eventDto.getTag());
+    }
 
-      if (existing != null) {
-        throw new CTPException(
-            CTPException.Fault.RESOURCE_VERSION_CONFLICT,
-            String.format(
-                "Event %s already exists for collection exercise %s",
-                eventDto.getTag(), collex.getId()));
-      } else {
-        Event event = new Event();
+    Event existing =
+        this.eventRepository.findOneByCollectionExerciseAndTag(collex, eventDto.getTag());
 
-        event.setCollectionExercise(collex);
-        event.setTag(eventDto.getTag());
-        event.setId(UUID.randomUUID());
-        event.setTimestamp(new Timestamp(eventDto.getTimestamp().getTime()));
-        event.setCreated(new Timestamp(new Date().getTime()));
+    if (existing != null) {
+      throw new CTPException(
+          CTPException.Fault.RESOURCE_VERSION_CONFLICT,
+          String.format(
+              "Event %s already exists for collection exercise %s",
+              eventDto.getTag(), collex.getId()));
+    }
 
-        event = eventRepository.save(event);
+    Event event = new Event();
 
-        fireEventChangeHandlers(CollectionExerciseEventPublisher.MessageType.EventCreated, event);
+    event.setCollectionExercise(collex);
+    event.setTag(eventDto.getTag());
+    event.setId(UUID.randomUUID());
+    event.setTimestamp(new Timestamp(eventDto.getTimestamp().getTime()));
+    event.setCreated(new Timestamp(new Date().getTime()));
 
-        return event;
-      }
+    createActionRulesForEvent(event, collex);
+    event = eventRepository.save(event);
+
+    fireEventChangeHandlers(CollectionExerciseEventPublisher.MessageType.EventCreated, event);
```

Comment:
MessageType should be imported properly and EventCreated should be EVENT_CREATED